src/Bicep.LangServer.UnitTests/Handlers/BicepEditLinterRuleCommandHandlerTests.cs

```diff
@@ -93,92 +94,39 @@ public class BicepEditLinterRuleCommandHandlerTests
             return selectedText;
         }
 
-        private class MockClientCapabilitiesProvider
+        private LanguageServerMock SetUpOnTriggerCompletion(LanguageServerMock server, Action<ShowDocumentParams> onShowDocument, Action<string?> onTriggerCompletion, bool enableShowDocumentCapability)
         {
-            public Mock<IClientCapabilitiesProvider> Mock;
-
-            public MockClientCapabilitiesProvider()
-            {
-                var clientCapabilitiesProvider = StrictMock.Of<IClientCapabilitiesProvider>();
-                clientCapabilitiesProvider.Setup(m => m.DoesClientSupportShowDocumentRequest()).Returns(true);
-                clientCapabilitiesProvider.Setup(m => m.DoesClientSupportWorkspaceFolders()).Returns(true);
-
-                Mock = clientCapabilitiesProvider;
-            }
-        }
-
-        private class LanguageServerMock
-        {
-            public Mock<ILanguageServerFacade> Mock;
-            public string? StringTriggeredForCompletion { get; private set; }
-            public ShowDocumentParams? ShowDocumentParams { get; private set; }
-
-            public LanguageServerMock(
-                ShowDocumentResult result,
-                Action<string>? messageCallback = null,
-                Container<WorkspaceFolder>? workspaceFolders = null)
-            {
-                var window = StrictMock.Of<IWindowLanguageServer>();
-
-                window
-                    .Setup(m => m.SendNotification(It.IsAny<LogMessageParams>()));
-                window
-                    .Setup(m => m.SendRequest<ShowDocumentResult>(It.IsAny<ShowDocumentParams>(), It.IsAny<CancellationToken>()))
-                    .Callback((IRequest<ShowDocumentResult> request, CancellationToken token) =>
+            ShowDocumentParams? showDocumentParams = null;
+            server.WindowMock.OnShowDocument(
+                p =>
+                {
+                    showDocumentParams = p;
+                    onShowDocument(p);
+                },
+                enableClientCapability: enableShowDocumentCapability);
+
+            server.Mock
+                .Setup(m => m.SendNotification("bicep/triggerEditorCompletion"))
```

Comment:
nit: Move "bicep/triggerEditorCompletion" to LangServerConstants and reuse it here and BicepEditLinterRuleCommandHandler?